docs/combine-queue.md

```diff
@@ -0,0 +1,224 @@
+example of computing phrasecounts using this recipe.
+
+ * TX1 queues `+1` update  for phrase `we want lambdas now`
+ * TX2 queues `+1` update  for phrase `we want lambdas now`
+ * TX3 reads the updates and current value for the phrase `we want lambdas now`.  There is no current value and the updates sum to 2, so a new value of 2 is written.
+ * TX4 queues `+2` update  for phrase `we want lambdas now`
+ * TX5 queues `-1` update  for phrase `we want lambdas now`
+ * TX6 reads the updates and current value for the phrase `we want lambdas now`.  The current value is 2 and the updates sum to 1, so a new value of 3 is written.
+
+Transactions processing updates have the ability to make additional updates.
+For example in addition to updating the current value for a phrase, the new
+value could also be placed on an export queue to update an external database.
+
+### Buckets
+
+A simple implementation of this recipe would have an update queue for each key.  However the
+implementation is slightly more complex.  Each update queue is in a bucket and transactions process
+all of the updates in a bucket.  This allows more efficient processing of updates for the following
+reasons :
+
+ * When updates are queued, notifications are made per bucket(instead of per a key).
+ * The transaction doing the update can scan the entire bucket reading updates, this avoids a seek for each key being updated.
+ * Also the transaction can request a batch lookup to get the current value of all the keys being updated.
+ * Any additional actions taken on update (like adding something to an export queue) can also be batched.
+ * Data is organized to make reading exiting values for keys in a bucket more efficient.
+
+Which bucket a key goes to is decided using hash and modulus so that multiple updates for a key go
+to the same bucket.
+
+The initial number of tablets to create when applying table optimizations can be controlled by
+setting the buckets per tablet option when configuring a Combine Queue.  For example if you
+have 20 tablet servers and 1000 buckets and want 2 tablets per tserver initially then set buckets
+per tablet to 1000/(2*20)=25.
+
+## Example Use
+
+The following code snippets show how to use this recipe for wordcount.  The first step is to
+configure it before initializing Fluo.  When initializing an ID is needed.  This ID is used in two
+ways.  First, the ID is used as a row prefix in the table.  Therefore nothing else should use that
+row range in the table.  Second, the ID is used in generating configuration keys.
+
+The following snippet shows how to configure a combine queue.
+
+```java
+    FluoConfiguration fluoConfig = ...;
+
+    // Set application properties for the combine queue.  These properties are read later by
+    // the observers running on each worker.
+    CombineQueue.configure(WcObserverProvider.ID)
+        .keyType(String.class).valueType(Long.class).buckets(119).finish(fluoConfig);
```

Comment:
I thought about save(), to me has a feeling of I/O.   What the method is doing is setting properties, I couldn't think of something short that would express this.  One good name is configure(), but then I don't know what to name the first method that starts the fluent chain.